src/core2/graph.test.js

```diff
@@ -71,6 +71,128 @@ describe("graph", () => {
       expect(() => graph.ref((null: any))).toThrow("null");
       expect(() => graph.ref((undefined: any))).toThrow("undefined");
     });
+    describe("neighbors", () => {
+      // Note: The tests share more state within this block than in the rest of the code.
+      // It should be fine so long as neighbors never mutates the graph.
+      const edge = (id, src, dst, type = "EDGE") => ({
+        address: {id, owner: {plugin: EXAMPLE_PLUGIN_NAME, type}},
+        src: src.address(),
+        dst: dst.address(),
+        payload: {},
+      });
+      // A cute little diagram:
+      // A>B = edge from A to B
+      // Nodes in the graph: B1,Foo
+      // (Parens) = Repeated or absent node
+      //
+      //    B1
+      //     V
+      //    Foo  > (Foo)
+      //     V
+      //   (B404)
+      const b1 = new BarPayload(1, "hello");
+      const b404 = new BarPayload(404, "hello");
+      const foo = new FooPayload();
+      const b1_foo = edge("b1_foo", b1, foo);
+      const foo_foo = edge("foo_foo", foo, foo, "SELF");
+      const foo_b404 = edge("foo_b404", foo, b404);
+      const phantomNode = new BarPayload(666, "ghost");
```

Comment:
The terms that we've been using so far are:
  - *present*: node is in the graph
  - *phantom*: node has been added to and removed from the graph (and is not in the graph)
  - *absent*: node was never in the graph
  - *isolated* (not trichotomous): node has no incident edges

So, your `phantomNode` isn't actually a phantom node. Could you rename it or update the graph structure accordingly?

Reply:
Done